app/assets/javascripts/angular/controllers/dashboard.js

```diff
@@ -49,30 +38,110 @@ dashboardController.controller('DashboardCtrl', ['$rootScope', '$scope', '$http'
       });
     }
 
+    // Make the News Slideshow work appropriately.
+    $scope.newsIndex = 0;
+
+    $scope.setCurrentSlideIndex = function (index) {
+        $scope.newsIndex = index;
+    };
+
+    $scope.isCurrentSlideIndex = function (index) {
+        return $scope.newsIndex === index;
+    };
+
+    $interval(function(){
+      $scope.newsIndex = ($scope.newsIndex < $scope.news.length - 1) ? ++$scope.newsIndex : 0;
+    }, 8000);
+
+    // Get the width of the Terror Tracker.
+    $scope.getTerrorWidth = function() {
+      var terror = $scope.terror;
+
+      // If Terror hasn't been retrieved yet, don't break.
+      if (terror === undefined) {
+        $scope.terror_width = 0;
+      } else if (terror > 250) {
+        $scope.terror = 250;
+        $scope.terror_width = 100;
+      } else {
+        $scope.terror_width = (100 / 250) * $scope.terror;
+      }
+
+      return {
+        'width': $scope.terror_width + '%'
+      };
+    };
+
+    // Get the colour of the Terror Tracker for the Thermometer Display.
+    $scope.getTerrorColour = function() {
+      // Check to see if the Terror amount is between two numbers.
+      function checkRange(x, n, m) {
+        if (x >= n && x <= m) { return x; }
+        else { return !x; }
+      }
+
+      var terror = $scope.terror;
+
+      // Set thermometer colour based on the Terror amount.
+      switch(terror) {
+        case checkRange(terror, 1, 50):
+          return'low';
+          break;
+        case checkRange(terror, 51, 100):
+          return 'med';
+          break;
+        case checkRange(terror, 101, 150):
+          return 'high';
+          break;
+        case checkRange(terror, 151, 200):
+          return 'crit';
+          break;
+        case checkRange(terror, 201, 250):
+          return 'doom';
+          break;
+        default:
+          return 'none';
+      }
+    };
+
     $scope.getStatus = function() {
       apiCall();
     };
 
-    $scope.updateNews = function(){
-      news_items = $('.news-container')
-      news_items.first().hide('slow',function(){
-          detach = news_items.first().detach()
-          detach.insertAfter(news_items.last()).fadeIn('slow');
-      });
-    }
-
     $scope.range = function(n) {
       return new Array(n);
     };
 
     $scope.getStatus();
 
-    $interval(function(){
-      $scope.updateNews();
-    }, 8000);
-
     $interval(function() {
       $scope.getStatus();
     }, 3000);
   }
-]);
+]).animation('.slide-animation', function () {
```

Comment:
It's for the news itself, not the stock ticker thing, which definitely means it needs comments.